### PR TITLE
Return the correct case from tables

### PR DESCRIPTION
**Pull request by dpeng817**
Return the correct case from fetch_last_updated_times

Changed test to ensure that table case is same going in as out.

### CODE REVIEW

**Comment by dpeng817** [comment]
* **#20984** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20984?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @dpeng817 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/libraries/dagster-snowflake/dagster_snowflake_tests/test_resources.py [hunk 1]
```diff
@@ -292,7 +292,7 @@ def test_fetch_last_updated_timestamps_empty():
         )
 
 
-@pytest.mark.skipif(not IS_BUILDKITE, reason="Requires access to the BUILDKITE snowflake DB")
+# @pytest.mark.skipif(not IS_BUILDKITE, reason="Requires access to the BUILDKITE snowflake DB")
```
Comment (johannkm) [inline python_modules/libraries/dagster-snowflake/dagster_snowflake_tests/test_resources.py:295]: why commented?

File: python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py [hunk 2]
```diff
@@ -740,8 +740,8 @@ def fetch_last_updated_timestamps(
         Mapping[str, datetime]: A dictionary of table names to their last updated time in UTC.
     """
     check.invariant(len(tables) > 0, "Must provide at least one table name to query upon.")
-    tables = [table.upper() for table in tables]
-    tables_str = ", ".join([f"'{table_name}'" for table_name in tables])
+    uppercase_tables = [table.upper() for table in tables]
```
Comment (johannkm) [inline python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py:743]: should fetch_last_updated_timestamps be marked experimental?

Comment (dpeng817) [reply python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py:743]: it should. Will do in another PR.

File: python_modules/libraries/dagster-snowflake/dagster_snowflake_tests/test_resources.py (hunk 1 above)
Comment (dpeng817) [reply python_modules/libraries/dagster-snowflake/dagster_snowflake_tests/test_resources.py:295]: commented to run the test locally. Will get rid of before landing.

File: python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py [hunk 3]
```diff
@@ -755,16 +755,16 @@ def fetch_last_updated_timestamps(
     if not result:
         raise ValueError("No results returned from Snowflake update time query.")
 
-    last_updated_times = {}
-    for table_name, last_altered in result:
+    result_mapping = {table_name: last_altered for table_name, last_altered in result}
```
Comment (johannkm) [inline python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py:758]: leave a comment that information_schema upper cases the table names

**Review by johannkm — APPROVED** [review]

**Pull request merged**